File: guess.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

int main() {
    srand(time(NULL));

    int number = (rand() % 100) + 1, guess = 0;

    std::cout << "guess a number between 1 and 100\n";

    while (guess != number) {
        std::cout << "> ";
        std::cin >> guess;

        if (guess < number) {
            std::cout << "too low!\n";
        }
        else if (guess > number) {
            std::cout << "too high!\n";
        }
    }
    std::cout << "that's right!\n";
    return 0;
}

File: guess.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	number := rand.Intn(100) + 1

	guess := 0

	fmt.Println("guess the number between 1 and 100")

	for guess != number {
		fmt.Print("> ")
		fmt.Scanln(&guess)

		if guess < number {
			fmt.Println("too low!")
		} else if guess > number {
			fmt.Println("too high!")
		}
	}
	fmt.Println("that's right!")
}
